### ISSUE TITLE
Unable to compile `scarb` when running `nile-rs run`

### ISSUE THREAD

**Issue by akirillo** [OpenZeppelin/nile-rs#35]
I'm trying to make use of `nile-rs` scripting, but am unable to run even the `example` script generated by `nile-rs init` due to the following errors:
```
error[E0277]: a value of type `HashMap<smol_str::SmolStr, PathBuf>` cannot be built from an iterator over elements of type `(SmolStr, _)`
    --> /Users/akirillo/.cargo/registry/src/github.com-1ecc6299db9ec823/scarb-0.1.0/src/compiler/helpers.rs:27:10
     |
27   |         .collect();
     |          ^^^^^^^ value of type `HashMap<smol_str::SmolStr, PathBuf>` cannot be built from `std::iter::Iterator<Item=(SmolStr, _)>`
     |
     = help: the trait `FromIterator<(SmolStr, _)>` is not implemented for `HashMap<smol_str::SmolStr, PathBuf>`
     = help: the trait `FromIterator<(K, V)>` is implemented for `HashMap<K, V, S>`
note: the method call chain might not have had the expected associated types
    --> /Users/akirillo/.cargo/registry/src/github.com-1ecc6299db9ec823/scarb-0.1.0/src/compiler/helpers.rs:21:10
     |
17   |       let crate_roots = unit
     |  _______________________-
18   | |         .components
     | |___________________- this expression has type `Vec<CompilationUnitComponent>`
19   |           .iter()
     |            ------ `Iterator::Item` is `&CompilationUnitComponent` here
20   |           .filter(|component| !component.package.id.is_core())
     |            --------------------------------------------------- `Iterator::Item` remains `&CompilationUnitComponent` here
21   |           .map(|component| {
     |  __________^
22   | |             (
23   | |                 component.cairo_package_name(),
24   | |                 component.target.source_root().into(),
25   | |             )
26   | |         })
     | |__________^ `Iterator::Item` changed to `(SmolStr, _)` here
note: required by a bound in `std::iter::Iterator::collect`
    --> /Users/akirillo/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/src/rust/library/core/src/iter/traits/iterator.rs:1887:19
     |
1887 |     fn collect<B: FromIterator<Self::Item>>(self) -> B
     |                   ^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Iterator::collect`

error[E0308]: mismatched types
  --> /Users/akirillo/.cargo/registry/src/github.com-1ecc6299db9ec823/scarb-0.1.0/src/compiler/helpers.rs:62:38
   |
62 |     vec![db.intern_crate(CrateLongId(unit.main_component().cairo_package_name()))]
   |                          ----------- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `smol_str::SmolStr`, found `SmolStr`
   |                          |
   |                          arguments to this struct are incorrect
   |
   = note: `SmolStr` and `smol_str::SmolStr` have similar names, but are actually distinct types
note: `SmolStr` is defined in crate `smol_str`
  --> /Users/akirillo/.cargo/registry/src/github.com-1ecc6299db9ec823/smol_str-0.1.24/src/lib.rs:36:1
   |
36 | pub struct SmolStr(Repr);
   | ^^^^^^^^^^^^^^^^^^
note: `smol_str::SmolStr` is defined in crate `smol_str`
  --> /Users/akirillo/.cargo/registry/src/github.com-1ecc6299db9ec823/smol_str-0.2.0/src/lib.rs:38:1
   |
38 | pub struct SmolStr(Repr);
   | ^^^^^^^^^^^^^^^^^^
   = note: perhaps two different versions of crate `smol_str` are being used?
note: tuple struct defined here
  --> /Users/akirillo/.cargo/registry/src/github.com-1ecc6299db9ec823/cairo-lang-filesystem-1.0.0-alpha.7/src/ids.rs:14:12
   |
14 | pub struct CrateLongId(pub SmolStr);
   |            ^^^^^^^^^^^

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `scarb` due to 2 previous errors
```

Which seems to stem from some conflict on the versions of `SmolStr` being used, looks like my cargo registry has both versions `0.1.24` and `0.2.0`. FWIW, I cleared my cargo registry (`rm -rf ~/.cargo/registry`) before executing this command, so I don't think the issue is with my system?

Other relevant context:
- I'm running `nile-rs` version `0.1.0`
- I have `scarb` separately installed as well, version `0.1.0 (289137c24 2023-03-28)`
- I'm trying to use Cairo 1 version `1.0.0-alpha.6` across my system, i.e. in my `PATH` there is a path to the `alpha.6` binary
  - This seems to be what my `scarb` installation recognizes, as `scarb --version` includes a line that says `cairo 1.0.0-alpha.6`
  - It seems that when I run `nile-rs run example`, it tries to separately compile Cairo 1 version `1.0.0-alpha.7` - not sure if this is relevant to the current issue, but also wary of that causing issues...
- I'm running macOS Ventura 13.3.1

**Comment by ericnordelo** [comment]
Hi @akirillo. Thanks for reporting this.

The issue is already fixed in [v0.1.0-rc.2](https://github.com/OpenZeppelin/nile-rs/releases/tag/v0.1.0-rc.2). You shouldn't have more issues if you build from this tag (and run `nile init` again).

**Explanation:**

The issue was coming from the last Cairo compiler release (`alpha.7`), and the way Scarb and us were managing the compiler dependencies.

Basically we were using:

```
[dependencies]
cairo-lang-starknet = "1.0.0-alpha.6"
```

instead of

```
[dependencies]
cairo-lang-starknet = { version = "1.0.0-alpha.6", git = "https://github.com/starkware-libs/cairo.git", rev = "439da05a" }
```

And rust handles these pre-releases changes as not breaking (from semver), so with the former, after the `alpha.7` release, this was the compiled version instead of the intended `alpha.6`.

**Comment by akirillo** [comment]
amazing! thank you.
